### ISSUE TITLE
SCA configuration blocks are not being merged

### ISSUE THREAD

**Issue by chemamartinez** [wazuh/wazuh#2755]
|Wazuh version|Install type|Install method|Platform|
|---|---|---|---|
| 3.9.0-3907 | Manager/Agent | All | All |

The SCA configuration is being overwritten by the last configuration block read, here we can see an example:

```
<sca>
    <enabled>yes</enabled>
    <scan_on_start>yes</scan_on_start>
    <interval>1m</interval>
    <skip_nfs>yes</skip_nfs>

    <policies>
      <policy>system_audit_rcl.yml</policy>
      <policy>system_audit_ssh.yml</policy>
      <policy>/var/ossec/etc/shared/cis_apache2224_rcl.yml</policy>
    </policies>
  </sca>

  ...

 <sca>
    <policies>
      <policy>cis_rhel7_linux_rcl.yml</policy>
    </policies>
  </sca>
```

When querying for the loaded configuration in that agent, we get the following parameters:

```
# curl -u foo:bar -k -X GET "http://127.0.0.1:55000/agents/001/config/wmodules/wmodules?pretty"
{
   "error": 0,
   "data": {
      "wmodules": [
         {
             {
            "sca": {
               "enabled": "yes",
               "scan_on_start": "yes",
               "skip_nfs": "yes",
               "interval": 43200,
               "policies": [
                  "cis_rhel7_linux_rcl.yml"
               ]
            }
         },
    ...
```

This is not the expected behavior. 

The binary options such as the `interval` must be overwritten only in the case they are set again explicitly. On the other hand, policies should be merged to be able to enable or disable them by the shared configuration.

### Update

This behavior affects all the modules present in Wazuh. There are two solutions proposed to solve this issue:

- The last configuration block overwrites the previous ones. This way, in the last block set, it is necessary to add every option we want to change from the default ones.
- Options with a specific value are not overwritten by the last one if are not set explicitly. Only the options which admit concatenated values are merged in the global configuration of that module.

### Update

For 3.9.0, option 2 of both exposed previously will be implemented for SCA and Syscollector.

**Comment by TJOSERAFAEL** [comment]
Hi @chemamartinez ,

this PR https://github.com/wazuh/wazuh/pull/2766 solves the issue.

Regards.

**Comment by chemamartinez** [comment]
Merged: #2766
